### DISCUSSION TITLE
How to access previous query's result within $transaction

### DISCUSSION THREAD

**Discussion by realmayus** [prisma/prisma#16070] [Q&A] [answered]
Assume the following transaction:
```ts
await prisma.$transaction([
  prisma.student.create({email: x, name: x});
  prisma.screening.create({studentId: ???, screenerId: 1})
]);
```
How can I access the result of the first `create` query (which contains a student's ID) from the second query, where I need to create a different object which relies on the student's ID?

**Comment by kyle-swanson** [comment, marked as answer]
I believe you will have to use the interactive transactions preview feature as described here:
[https://www.prisma.io/docs/concepts/components/prisma-client/transactions#interactive-transactions-in-preview](https://www.prisma.io/docs/concepts/components/prisma-client/transactions#interactive-transactions-in-preview
)

**Reply by pantharshit00** [reply]
Yeah, you will need to use interactive transaction here. It will allow you to consume intermediate results while running all queries in a single transaction.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
